### PR TITLE
rsx capture: Fix regression after #7730

### PR DESCRIPTION
**Pull request by elad335**
* Fix rsx capture state after fmt::throw_exception.
* Disable capture button if no game is running!
* Fix UB/race in rsx captures. (unprotected variables are read and written concurrently)

### CODE REVIEW

**Comment by elad335** [comment]
Added a yakuza fix. (#7330 rebroke #5713 which is fixed by this pr)

File: rpcs3/Emu/RSX/RSXThread.cpp [hunk 1]
```diff
@@ -295,6 +295,8 @@ namespace rsx
 	thread::~thread()
 	{
 		g_access_violation_handler = nullptr;
+		user_asked_for_frame_capture = false;
```
Comment (kd-11) [inline rpcs3/Emu/RSX/RSXThread.cpp]: This seems to be a workaround against a real bug.

File: Utilities/Thread.cpp [hunk 2]
```diff
@@ -2024,6 +2025,11 @@ void thread_ctrl::emergency_exit(std::string_view reason)
 
 	if (const auto _this = g_tls_this_thread)
 	{
+		while (_this->m_state != thread_state::aborting)
```
Comment (Nekotekina) [inline Utilities/Thread.cpp]: It'll freeze some threads forever. I'd like to avoid it.

File: Utilities/Thread.cpp [hunk 3]
```diff
@@ -1450,6 +1450,7 @@ bool handle_access_violation(u32 addr, bool is_writing, x64_context* context) no
 			{
 				vm_log.notice("\n%s", cpu->dump());
 				vm_log.fatal("Access violation %s location 0x%x (%s)", is_writing ? "writing" : "reading", addr, (is_writing && vm::check_addr(addr)) ? "read-only memory" : "unmapped memory");
+				Emu.Resume(); // Special case
```
Comment (Nekotekina) [inline Utilities/Thread.cpp]: Changed severity to "error"

**Comment by elad335** [comment]
Improved the rsx capture fox.

**Pull request merged**